tezos/messages/src/p2p/encoding/peer.rs

```diff
@@ -47,74 +49,80 @@ pub struct PeerMessageResponse {
 
 cached_data!(PeerMessageResponse, body);
 has_encoding!(PeerMessageResponse, PEER_MESSAGE_RESPONSE_ENCODING, {
-    Encoding::Obj(vec![Field::new(
-        "messages",
-        Encoding::dynamic(Encoding::list(Encoding::Tags(
-            size_of::<u16>(),
-            TagMap::new(vec![
-                Tag::new(0x01, "Disconnect", Encoding::Unit),
-                Tag::new(0x02, "Bootstrap", Encoding::Unit),
-                Tag::new(0x03, "Advertise", AdvertiseMessage::encoding().clone()),
-                Tag::new(0x04, "SwapRequest", SwapMessage::encoding().clone()),
-                Tag::new(0x05, "SwapAck", SwapMessage::encoding().clone()),
-                Tag::new(
-                    0x10,
-                    "GetCurrentBranch",
-                    GetCurrentBranchMessage::encoding().clone(),
-                ),
-                Tag::new(
-                    0x11,
-                    "CurrentBranch",
-                    CurrentBranchMessage::encoding().clone(),
-                ),
-                Tag::new(0x12, "Deactivate", DeactivateMessage::encoding().clone()),
-                Tag::new(
-                    0x13,
-                    "GetCurrentHead",
-                    GetCurrentHeadMessage::encoding().clone(),
-                ),
-                Tag::new(0x14, "CurrentHead", CurrentHeadMessage::encoding().clone()),
-                Tag::new(
-                    0x20,
-                    "GetBlockHeaders",
-                    GetBlockHeadersMessage::encoding().clone(),
-                ),
-                Tag::new(0x21, "BlockHeader", BlockHeaderMessage::encoding().clone()),
-                Tag::new(
-                    0x30,
-                    "GetOperations",
-                    GetOperationsMessage::encoding().clone(),
-                ),
-                Tag::new(0x31, "Operation", OperationMessage::encoding().clone()),
-                Tag::new(
-                    0x40,
-                    "GetProtocols",
-                    GetProtocolsMessage::encoding().clone(),
-                ),
-                Tag::new(0x41, "Protocol", ProtocolMessage::encoding().clone()),
-                Tag::new(
-                    0x50,
-                    "GetOperationHashesForBlocks",
-                    GetOperationHashesForBlocksMessage::encoding().clone(),
-                ),
-                Tag::new(
-                    0x51,
-                    "OperationHashesForBlocks",
-                    OperationHashesForBlocksMessage::encoding().clone(),
-                ),
-                Tag::new(
-                    0x60,
-                    "GetOperationsForBlocks",
-                    GetOperationsForBlocksMessage::encoding().clone(),
-                ),
-                Tag::new(
-                    0x61,
-                    "OperationsForBlocks",
-                    OperationsForBlocksMessage::encoding().clone(),
+    Encoding::bounded(
```

Comment:
@akoptelov this is the thing we discussed some time ago about a mismatch between our implementation representing this as a list while in OCaml it was a single dynamically-sized element, is that correct?

Reply:
What I am thinking is, why does this need to be a list at all? (I am not that familiar with our encoding library, so there may be details that I am missing)

Reply:
Well, just limiting the list a very fast fix, with minimal changes. But I agree, replacing the list with single value is more clear and firm solution.